tasks/RedHat/installation.yml

```diff
@@ -78,44 +70,85 @@
-      loop_control:
-        loop_var: pkg
-    - debug: var=sap_hana_preconfigure_fact_minpkglist
-
-    - name: Install minimum packages if required
-      yum:
-        name: "{{  sap_hana_preconfigure_fact_minpkglist }}"
-        state: present
-      when: not ( sap_hana_preconfigure_fact_minpkglist == [ "" ] )
+  - name: Gather package facts
+    package_facts:
+
+  - set_fact:
+      reqpkgs: "{{ __sap_hana_preconfigure_min_pkgs }}"
+
+  - debug:
+      var: reqpkgs
+
+  - set_fact:
+      sap_hana_preconfigure_fact_minpkglist: |-
+        {% set ns = namespace() %}
+        {% set ns.pkgs = [] %}
+        {% for reqpkg in reqpkgs %}
+        {%   if not reqpkg[0] in ansible_facts.packages %}
+        {%     set _ = ns.pkgs.append(reqpkg | join('-')) %}
+        {%   else %}
+        {%     set pkg = ansible_facts.packages[reqpkg[0]][-1] %}
+        {%     set verrel = pkg.version ~ '-' ~ pkg.release %}
+        {%     if reqpkg[1].replace (".el", ".0.0.") is version(verrel.replace (".el", ".0.0."), '>', strict=false) %}
+        {%       set _ = ns.pkgs.append(reqpkg | join('-')) %}
+        {%     else %}
+        {%       set _ = "pkg already installed with same or later version" %}
+        {%     endif %}
+        {%   endif %}
+        {% endfor %}
+        {{ ns.pkgs }}
+
+#    - name: check if minimum release needs to be installed
+#      shell: |
+#        set -x
+#        required_pkg={{ pkg | join('-') }}
+#        newest=$(echo -e "$required_pkg\n$(rpm -q {{ pkg[0] }} )"| sed "s,\.el,.0.0.," | sort -V | tail -1)
+#        if [ $newest == ${required_pkg/.el/.0.0.} ]; then
+#           echo $newest
+#        fi
+#      loop: "{{ __sap_hana_preconfigure_min_pkgs }}"
+#      loop_control:
+#        loop_var: pkg
+#      check_mode: no
+#      register: sap_hana_preconfigure_register_minpkglist
+#      changed_when: false
+
+#    - name: Initialize an empty list for our strings
+#      set_fact:
+#        sap_hana_preconfigure_fact_minpkglist: []
+
+#    - name: Create list of packages to be installed
+#      set_fact:
+#        sap_hana_preconfigure_fact_minpkglist: "{{ sap_hana_preconfigure_fact_minpkglist | difference(['']) + [ pkg.stdout ] }}"
+#      loop: "{{ sap_hana_preconfigure_register_minpkglist.results }}"
+#      loop_control:
+#        loop_var: pkg
+
+  - debug: var=sap_hana_preconfigure_fact_minpkglist
+
+  - name: Install minimum packages if required
+    yum:
+      name: "{{ sap_hana_preconfigure_fact_minpkglist }}"
+      state: present
+    when: not ( sap_hana_preconfigure_fact_minpkglist == [ "" ] )
 
   when: 
     - sap_hana_preconfigure_min_package_check|bool
-    - not( (sap_hana_preconfigure_min_pkgs is undefined) or (sap_hana_preconfigure_min_pkgs is none) or (sap_hana_preconfigure_min_pkgs | trim == '') ) 
+    - not( (__sap_hana_preconfigure_min_pkgs is undefined) or (__sap_hana_preconfigure_min_pkgs is none) or (__sap_hana_preconfigure_min_pkgs | trim == '') ) 
```

Comment:
you could use `when: __sap_hana_preconfigure_min_pkgs | d([])`
unless you need to do various types of input validation, if the user mistakenly sets `__sap_hana_preconfigure_min_pkgs: " "`, or other user errors

Reply:
makes sense to me.